unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -4547,66 +4538,90 @@ protected SimObjPhysics[] GetAllVisibleSimObjPhysics(
-            } else {
-                return GetAllVisibleSimObjPhysicsDistance(
-                    camera,
-                    maxDistance,
-                    filterSimObjs,
-                    out interactable
-                );
-            }
+            return GetAllVisibleSimObjPhysicsDistance(
+                camera,
+                maxDistance,
+                filterSimObjs,
+                out interactable
+            );
         }
 
-        protected VisibilityScheme getVisibilityScheme(string visibilityScheme = null) {
-            VisibilityScheme visSchemeEnum;
-            if (visibilityScheme != null) {
-                visibilityScheme = visibilityScheme.ToLower();
-
-                if (
-                    visibilityScheme
-                    == Enum.GetName(typeof(VisibilityScheme), VisibilityScheme.Collider).ToLower()
-                ) {
-                    visSchemeEnum = VisibilityScheme.Collider;
-                } else if (
-                      visibilityScheme
-                      == Enum.GetName(typeof(VisibilityScheme), VisibilityScheme.Distance).ToLower()
-                  ) {
-                    visSchemeEnum = VisibilityScheme.Distance;
-                } else {
+        //return a list of visible sim objects from a specific camera
+        // maxDistance defaults to whatever initialized by agent's maxVisibleDistance
+        // thirdPartyCameraId defaults to use main camera
+        // filterObjectIds defaults to null, which means all objects are considered
+        // note: visible objects are returned but this does not guarantee the objects are interactable (ex: something behind glass)
+        public void GetVisibleObjectsFromCamera(
+            float? maxDistance = null, //max distance from the camera origin to consider objects visible
+            int? thirdPartyCameraId = null, //leave null to query main camera, otherwise pass in the index of the third party camera
+            List<string> filterObjectIds = null //only return objects with these ids
+        ) {
+            //if thirdPartyCameraId not specified, default to main camera
+            Camera camera = m_Camera;
+            if (thirdPartyCameraId.HasValue) {
+                camera = agentManager.thirdPartyCameras[thirdPartyCameraId.Value];
+                if (this.visibilityScheme != VisibilityScheme.Distance) {
                     throw new System.NotImplementedException(
-                        $"Visibility scheme {visibilityScheme} is not implemented. Must be 'distance' or 'collider'."
+                        $"Visibility scheme {this.visibilityScheme} is not implemented for third party cameras. Default visibility scheme should be 'Distance'."
                     );
                 }
-            } else {
-                visSchemeEnum = this.visibilityScheme;
             }
-            return visSchemeEnum;
+
+            //only check visibility for objects with these ids otherwise check them all
```

Comment:
"otherwise check them all" has no code-based follow-through. However, I'm not sure you'd ever want to check everything, since it goes against the entire purpose of the function. For that reason, I'd alter the comment rather than the code.